npe2/manifest/schema.py

```diff
@@ -282,14 +291,16 @@ def discover(cls, entry_point_group=ENTRY_POINT) -> Iterator[PluginManifest]:
                 try:
                     pm = cls._from_entrypoint(ep)
                     pm._populate_missing_meta(dist.metadata)
-                    yield pm
-                except ValidationError:
+                    yield pm, None
+                except ValidationError as e:
                     logger.warning(msg=f"Invalid schema {ep.value!r}")
+                    yield None, e
                 except Exception as e:
                     logger.warning(
                         "%s -> %r could not be imported: %s"
                         % (entry_point_group, ep.value, e)
                     )
+                    yield None, e
```

Comment:
it would be nice here, and above, to associate the entry point object `ep` as well.   should it be a tuple of `(e, ep)`?

Reply:
+1.  I wouldn't mind having a type encapsulating `(e,ep)` to make it a little easier on the type signatures.